cfme/tests/cloud_infra_common/test_power_control_rest.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+
+from cfme.common.vm import VM
+from utils import testgen
+from utils.version import current_version
+from utils.wait import wait_for
+
+
+pytestmark = [
+    pytest.mark.usefixtures('uses_infra_providers', 'uses_cloud_providers')
+]
+
+
+def pytest_generate_tests(metafunc):
+    # Filter out providers without provisioning data or hosts defined
+    argnames, argvalues, idlist = testgen.all_providers(
+        metafunc, 'small_template', template_location=["small_template"])
+
+    testgen.parametrize(metafunc, argnames, argvalues, ids=idlist, scope="module")
+
+
+@pytest.fixture(scope="function")
+def vm_obj_and_rest(request, provider, setup_provider, small_template, rest_api):
+    vm_obj = VM.factory(
+        'test_pwrctl_{}'.format(fauxfactory.gen_alpha(length=8).lower()),
+        provider, template_name=small_template)
+
+    @request.addfinalizer
+    def _delete_vm():
+        if provider.mgmt.does_vm_exist(vm_obj.name):
+            provider.mgmt.delete_vm(vm_obj.name)
+    vm_obj.create_on_provider(find_in_cfme=True, allow_skip="default")
+    if provider.category == 'cloud':
+        vm = rest_api.collections.instances.get(name=vm_obj.name)
+    elif provider.category == 'infra':
+        vm = rest_api.collections.vms.get(name=vm_obj.name)
+    else:
+        raise Exception("Unknown provider type")
```

Comment:
Currently it's workaround to get instance or vm via rest. I'd rather add method get_vm_via_rest to VM class and overload implementation for instance/vm. What do you think @psav @mfalesni ?